src/notation/internal/masternotation.cpp

```diff
@@ -625,23 +625,17 @@ void MasterNotation::updateExcerpts()
 {
     ExcerptNotationList newExcerpts;
 
-    for (IExcerptNotationPtr excerpt : m_excerpts.val) {
-        if (!get_impl(excerpt)->excerpt()->isEmpty()) {
-            newExcerpts.push_back(excerpt);
-        }
-    }
-
     QList<Ms::Excerpt*> excerpts = score()->excerpts();
 
     for (Part* part: score()->parts()) {
-        bool isNewPart = true;
-
-        for (Ms::Excerpt* exceprt: excerpts) {
-            isNewPart &= !exceprt->containsPart(part);
-        }
+        auto existing = std::find_if(m_excerpts.val.begin(), m_excerpts.val.end(), [part](auto e) {
+            return get_impl(e)->excerpt()->containsPart(part);
+        });
 
-        if (isNewPart) {
+        if (existing == m_excerpts.val.end()) {
             newExcerpts.push_back(createExcerpt(part));
+        } else if (std::find(newExcerpts.begin(), newExcerpts.end(), *existing) == newExcerpts.end()) {
+            newExcerpts.push_back(*existing);
```

Comment:
For each part, find first existing excerpt containing it - if there isn't one, create a new one and add to list, otherwise if existing one not already added to list, add it.
However...I don't know that this will correctly re-order the parts *within* excerpts that contain multiple parts, I got weird exceptions trying to create such excerpts (not related to this change, they happen on master too).